libs/ffi/ffi.c

```diff
@@ -240,6 +306,39 @@ int FFI_mitls_connect(struct _FFI_mitls_callbacks *callbacks, /* in */ mitls_sta
     CAMLreturnT(int,ret);
 }
 
+// Called by the host server app, after a client has connected to a socket and the calling server has accepted the TCP connection.
+int FFI_mitls_accept_connected(struct _FFI_mitls_callbacks *callbacks, /* in */ mitls_state *state, /* out */ char **outmsg, /* out */ char **errmsg)
+{
+    CAMLparam0();
+    CAMLlocal1(result);
+    int ret;
+
+    *outmsg = NULL;
+    *errmsg = NULL;
+
+    caml_acquire_runtime_system();
+    result = caml_callback2_exn(*g_mitls_FFI_AcceptConnected, state->fstar_state, PtrToValue(callbacks));
+    if (Is_exception_result(result)) {
+        // Call caml_format_exception(Extract_exception(result)) to extract the exception text
+        ret = 0;
+    } else {
+        // AcceptConnected returns back (Connection.connection * int)
+        value connection = Field(result,0);
+        ret = Int_val(Field(result,1));
+        if (ret == 0) {
+            caml_modify_generational_global_root(&state->fstar_state, connection);
```

Comment:
I have a doubt about this line because the mutable field is called `state` but it is replaced by `connection`. Probably just needs a comment, e.g. "the semantics of AcceptConnected is that it returns a Connection.connection, along with an integer that tells whether the new connection was successfully accepted or not".

- Why is `AcceptedConnected` not returning a `connection option`?
- Also, what happens, memory-wise, with the previously-allocated OCaml value that was in `state->fstar_state`? I assume it is collected by the OCaml GC?